### DISCUSSION TITLE
safelist error with double underscore

### DISCUSSION THREAD

**Discussion by charliezstong** [tailwindlabs/tailwindcss#7431] [General]
What version of Tailwind CSS are you using?
3.0.19

What build tool (or framework if it abstracts the build tool) are you using?
none (pug,js,css)

What version of Node.js are you using?
v14.17.1

What browser are you using?
Chrome

What operating system are you using?
Windows 10

Describe your issue
The safelist pattern not working with intlTelInput.css ( https://github.com/jackocnr/intl-tel-input/blob/master/build/css/intlTelInput.css  ) can't extract selector with double underscore "__". 

`
safelist:[
{pattern: /^iti/, variants: ['__']}
]
`

**Comment by adamwathan** [comment]
Hey! Not sure what you're trying to do here, `'__'` is not a variant so I'm not sure why it's in the variants array. Variants are things like `hover`, `focus`, etc. Adding `__` as a variant there implies that you have classes like `__:iti` which I don't see in your CSS file.

If you can share a project where you actually have this setup to the best of your ability it will be easier to see where things are going wrong and point you in the right direction.

**Reply by charliezstong** [reply]
Hi @adamwathan, check this: https://play.tailwindcss.com/5e3xhHm453 (my project setup)

`.iti` works, extracts some selectors...

But classes that contain double underscores as an `iti__flag` do not extract the selector.
Change in the CSS of the library the double underscore for --  `(iti--flag)` and there if I extracted the classes.

I would like to understand this error, I am not sure if I am doing something wrong.
